### PR TITLE
tsdb Use testing.T.TempDir() instead of ioutil.TempDir() in unit tests

### PR DESCRIPTION
**Pull request by praveenghuge**
Related to #9594
small PR to fix tsdb directory

### CODE REVIEW

**Comment by roidelapluie** [comment]
Can you please rebase this pull request? thank you

File: tsdb/db_test.go [hunk 1]
```diff
@@ -58,16 +58,15 @@ func TestMain(m *testing.M) {
...
 	if len(rngs) == 0 {
-		db, err = Open(tmpdir, nil, nil, opts, nil)
+		db, _ = Open(tmpdir, nil, nil, opts, nil)
 	} else {
 		opts, rngs = validateOpts(opts, rngs)
-		db, err = open(tmpdir, nil, nil, opts, rngs, nil)
+		db, _ = open(tmpdir, nil, nil, opts, rngs, nil)
 	}
-	require.NoError(t, err)
+	require.NoError(t, nil)
```
Comment (codesome) [inline tsdb/db_test.go:69]: We should not ignore the error here. You can have `var err error` before the `if` and check the error.

File: tsdb/db_test.go [hunk 2]
```diff
@@ -2364,14 +2310,7 @@ func TestDBReadOnly_FlushWAL(t *testing.T) {
...
 	require.NoError(t, err)
 
-	flush, err := ioutil.TempDir("", "flush")
-	require.NoError(t, err)
-
-	defer func() {
-		require.NoError(t, os.RemoveAll(flush))
-	}()
-	require.NoError(t, db.FlushWAL(flush))
-	require.NoError(t, db.Close())
```
Comment (codesome) [inline tsdb/db_test.go:2374]: Removed by mistake?

File: tsdb/querier_test.go [hunk 3]
```diff
@@ -1400,7 +1390,7 @@ func BenchmarkQuerySeek(b *testing.B) {
 					require.NoError(b, it.Err())
 				}
 				require.NoError(b, ss.Err())
-				require.NoError(b, err)
+				require.NoError(b, nil)
```
Comment (codesome) [inline tsdb/querier_test.go:1393]: You can remove this line now

**Review by codesome — CHANGES_REQUESTED** [review]
Thanks for this PR! We have to fix few erroneous changes before we can merge :)

**Comment by beorn7** [comment]
FTR: @milencium wants to work on #9594, which might be based on this PR or might be a complete restart.

**Pull request closed without merging**